### DISCUSSION TITLE
compiling from source

### DISCUSSION THREAD

**Discussion by lunaseraphica** [PlaceholderAPI/PlaceholderAPI#955] [(2) Support] [answered]
Yo, how do i compile this to use in my maven project for building my plugin?.. i don't see any pom.xml file in the repo so that's why i am asking

**Comment by Andre601** [comment, marked as answer]
It's a Gradle project, so if you want to build it, you have to use Gradle instead of Maven for PlaceholderAPI.
The command should be `gradlew clean shadowJar` (or `./gradlew clean shadowJar` if you're on Windows).

Tho, if your goal is to hook into PlaceholderAPI is it highly recommended to use the external repository instead of using jar files.
A starter guide is found here: https://github.com/PlaceholderAPI/PlaceholderAPI/wiki/Hook-into-PlaceholderAPI
